### ISSUE TITLE
binascii.Error: Incorrect padding / paramiko.hostkeys.InvalidHostKey error with v1.4.17 (but not 1.4.14)

### ISSUE THREAD

**Issue by mjmare** [Fizzadar/pyinfra#683]
Hi 

I'm getting the error below. When reverting to the previous PyInfra version (from 1.4.17 to 1.4.14), everything worked again. 
I have recreated both virtualenvs and this behaviour is reproducible. 

```
--> Connecting to hosts...
Traceback (most recent call last):
  File "/Users/mjm/Dropbox/Devel/ACHN/Anvil_InUit/venv/lib/python3.9/site-packages/paramiko/hostkeys.py", line 360, in from_line
    key = RSAKey(data=decodebytes(key))
  File "/Users/mjm/.pyenv/versions/3.9.6/lib/python3.9/base64.py", line 538, in decodebytes
    return binascii.a2b_base64(s)
binascii.Error: Incorrect padding

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 906, in gevent._gevent_cgreenlet.Greenlet.run
  File "/Users/mjm/Dropbox/Devel/ACHN/Anvil_InUit/venv/lib/python3.9/site-packages/pyinfra/api/host.py", line 159, in connect
    self.connection = self.executor.connect(self.state, self)
  File "/Users/mjm/Dropbox/Devel/ACHN/Anvil_InUit/venv/lib/python3.9/site-packages/pyinfra/api/connectors/ssh.py", line 210, in connect
    client.load_system_host_keys()
  File "/Users/mjm/Dropbox/Devel/ACHN/Anvil_InUit/venv/lib/python3.9/site-packages/paramiko/client.py", line 104, in load_system_host_keys
    self._system_host_keys.load(filename)
  File "/Users/mjm/Dropbox/Devel/ACHN/Anvil_InUit/venv/lib/python3.9/site-packages/paramiko/hostkeys.py", line 101, in load
    e = HostKeyEntry.from_line(line, lineno)
  File "/Users/mjm/Dropbox/Devel/ACHN/Anvil_InUit/venv/lib/python3.9/site-packages/paramiko/hostkeys.py", line 372, in from_line
    raise InvalidHostKey(line, e)
paramiko.hostkeys.InvalidHostKey: ('fe80::a7b5:90f6:dc54:3028%en1 ssh-rsa AAAAB3NzaC1yc2EAAAADAQAB.......q3+tYmCTlo8bW3GNm9Y=', Error('Incorrect padding'))
2021-10-26T09:35:48Z <Greenlet at 0x102bb5ae0: <bound method Host.connect of Host(toolserver.xxxx.yyyy)>> failed with InvalidHostKey

--> An unexpected internal exception occurred:

  File "/Users/mjm/Dropbox/Devel/ACHN/Anvil_InUit/venv/lib/python3.9/site-packages/paramiko/hostkeys.py", line 372, in from_line
    raise InvalidHostKey(line, e)
Traceback (most recent call last):
  File "/Users/mjm/Dropbox/Devel/ACHN/Anvil_InUit/venv/lib/python3.9/site-packages/paramiko/hostkeys.py", line 360, in from_line
    key = RSAKey(data=decodebytes(key))
  File "/Users/mjm/.pyenv/versions/3.9.6/lib/python3.9/base64.py", line 538, in decodebytes
    return binascii.a2b_base64(s)
binascii.Error: Incorrect padding

During handling of the above exception, another exception occurred:

paramiko.hostkeys.InvalidHostKey: ('fe80::a7b5:90f6:dc54:3028%en1 ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAA....... bW3GNm9Y=', Error('Incorrect padding'))

--> The full traceback has been written to pyinfra-debug.log
--> If this is unexpected please consider submitting a bug report on GitHub, for more information run `pyinfra --support`.
error: Recipe `deploy` failed on line 8 with exit code 1
```

```
 pyinfra --support
--> Support information:

    If you are having issues with pyinfra or wish to make feature requests, please
    check out the GitHub issues at https://github.com/Fizzadar/pyinfra/issues .
    When adding an issue, be sure to include the following:

    System: Darwin
      Platform: macOS-11.6-x86_64-i386-64bit
      Release: 20.6.0
      Machine: x86_64
    pyinfra: v1.4.17
    Executable: /Users/mjm/Dropbox/Devel/ACHN/Anvil_InUit/venv/bin/pyinfra
    Python: 3.9.6 (CPython, Clang 12.0.5 (clang-1205.0.22.11))
```

**Comment by Fizzadar** [comment]
So annoyingly this looks like a bad key entry in your host keys - it's not ideal that this throws loading the rest of them. In v1.4.14 the host keys weren't being loaded at all which was a bug fixed, but has caused multiple issues since. Known host files get manipulated a lot so I'm not surprised.

I'm going to capture errors loading them and display a warning in that situation, as it shouldn't throw pyinfra entirely.

**Comment by mjmare** [comment]
Could there be a bad entry if the key functions normally?

**Comment by Fizzadar** [comment]
I believe so - if you connect to that host does SSH show any errors/warnings? And are there any other keys for the same host?

AFAIK SSH will simply ignore badly formatted lines, whereas paramiko seems to bail right away.

**Comment by mjmare** [comment]
When I connect manually I don't get any errors.

**Comment by Fizzadar** [comment]
I've just added a "fix" for errors encountered when loading SSH keys in https://github.com/Fizzadar/pyinfra/commit/52beef021b767cd259c80da137befa49de1c4456 and released in `v1.4.18`, which should prevent this failing going forward. The underlying issue needs to be fixed either in paramiko or your system host keys.
